R6 user login

DeeDoubleYous opened the pull request:
Includes the current implementation of the user login. CSS styling needed to be complete, but functionally should be complete. After this save for later can be done. 
Issues with this would be with storing passwords in plain text and not doing any fancy encryption at all but in this "fake it" scenario we can  live with it.

DeeDoubleYous committed:
original face for user login + web.xml change for navigation

DeeDoubleYous committed:
put the email and passwords inputs in a table

DeeDoubleYous committed:
initial user login controller/user entity prototype, removed table, can be styled in css later

DeeDoubleYous committed:
an attempt to make the login control actually interact with the database

DeeDoubleYous committed:
Make changes to reflect the updated ERD

DeeDoubleYous committed:
has a very prototype look up. not close to ready

DeeDoubleYous committed:
small changes to move to other database while main one settles

DeeDoubleYous committed:
Merge branch 'master' into R6_Users_and_save_for_later

DeeDoubleYous committed:
clean up from merge conflicts

DeeDoubleYous committed:
fix redirect on the login control

DeeDoubleYous committed:
fix the redirect in the trader class

DeeDoubleYous committed:
Merge branch 'master' into R6_Users_and_save_for_later

DeeDoubleYous committed:
changed password input to secret

mc1098 commented on src/main/java/bton/ci536/fizzit/trade/LocalTradeList.java:
```diff
@@ -30,6 +31,8 @@
     private double totalValue = 0;
     private int totalItems = 0;
     
+    private Customer customer;
```
Why do you need Customer here?

DeeDoubleYous replied:
Oh that was for something I was going to attempt. But it's not needed now. Will get rid of when I get home.

mc1098 commented on src/main/java/bton/ci536/fizzit/trade/LocalTradeList.java:
```diff
@@ -130,4 +133,12 @@ public void clear(){
         items.clear();
     }
+
+	public Customer getCustomer() {
+		return customer;
+	}
+
+	public void setCustomer(Customer customer) {
+		this.customer = customer;
+	}
```
As above

DeeDoubleYous replied:
Same as above

mc1098 commented on src/main/webapp/index.xhtml:
```diff
@@ -14,6 +14,7 @@
     </h:head>
     <h:body>
         
+        <h:outputText styleClass="log-out" value="#{LocalTradeList.customerId}"/>
```
This renders an empty span element for me above the menu. 
Would this be better in the menu component? could go in the drop-down menu after "trades"?

mc1098 commented on src/main/webapp/resources/css/style.css:
```diff
@@ -213,9 +213,6 @@ main{
     opacity: 0.7;
 }
 
-.user-sec{
-    color: #fff;
-}
```
Removing this makes the user name in the menu black.

DeeDoubleYous replied:
I think that maybe a victim of when I was merging your stuff into this branch. Not intended to be removed. Will reinstate.

mc1098 reviewed (changes requested):
Just needs some minor changes - inputSecret is a good change. 
Sorry for delay in reviewing this.

Commit:
Correction to incorrect logic from a previous commit of mine. 
Next should only return the same status if the current status is completed 
or above.

DeeDoubleYous committed:
remove unneeded customer references in tradeList

DeeDoubleYous committed:
removed the import of customer

DeeDoubleYous committed:
replaced the CSS for the username

DeeDoubleYous committed:
Implemented a log out feature, removed unused log out outputText

DeeDoubleYous committed:
Merge branch 'R6_Users_and_save_for_later' of https://github.com/DA-Wishart/Fizzit.com into R6_Users_and_save_for_later

mc1098 commented on src/main/webapp/resources/comp/menu.xhtml:
```diff
@@ -19,9 +21,15 @@
+                        				class="log-out-btn"
+                        				value="Log out"
+                        				action="#{loginControl.logOut(customer)}">
+                        	<f:ajax execute="@menu" resetValues="true" render="@menu"/>
```
Do you need ajax here? the loginControl::logOut function is going to redirect the page anyway correct?

DeeDoubleYous replied:
it was from the experimentation previously. Its my bad habit to leave that sort of stuff in there

mc1098 reviewed (commented):
Just the ajax thing I'm not sure about, does it cause an issue without it?

DeeDoubleYous committed:
Remove uneeded ajax, left over from experimentation
